src/integration-test/java/com/mongodb/jdbc/integration/testharness/IntegrationTestUtils.java

```diff
@@ -0,0 +1,428 @@
+            throws SQLException, InvocationTargetException, IllegalAccessException {
+
+        List<Object> metadataFunction = entry.meta_function;
+        assertTrue(metadataFunction != null && metadataFunction.size() > 0);
+        String functionName = (String) metadataFunction.remove(0);
+        Method[] m = DatabaseMetaData.class.getMethods();
+
+        for (Method method : m) {
+            if (method.getName().equalsIgnoreCase(functionName)) {
+                Class<?>[] types = method.getParameterTypes();
+                if (method.getReturnType() != ResultSet.class) {
+                    throw new IllegalArgumentException(
+                            "expected: "
+                                    + ResultSet.class.getName()
+                                    + " found: "
+                                    + method.getReturnType());
+                }
+                if (method.getParameterCount() != metadataFunction.size()) {
+                    throw new IllegalArgumentException(
+                            "expected parameter count: "
+                                    + method.getParameterCount()
+                                    + " found: "
+                                    + metadataFunction.size());
+                }
+                if (method.getParameterCount() == 0) {
+                    return (ResultSet) method.invoke(databaseMetaData);
+
+                } else {
+                    Object[] parameters = new Object[method.getParameterCount()];
+
+                    for (int i = 0; i < parameters.length; i++) {
+                        Object currentMetaInput = metadataFunction.get(i);
+                        if (currentMetaInput == null) {
+                            parameters[i] = null;
+                            continue;
+                        }
+                        Class<?> parameterType = types[i];
+                        if (parameterType.isArray()) {
+                            if (parameterType.getComponentType() == String.class) {
+                                parameters[i] =
+                                        ((List<String>) currentMetaInput)
+                                                .stream()
+                                                .map(object -> Objects.toString(object, null))
+                                                .toArray(String[]::new);
+                            } else if (parameterType.getComponentType() == int.class) {
+                                parameters[i] =
+                                        ((List<Integer>) currentMetaInput)
+                                                .stream()
+                                                .mapToInt(j -> j)
+                                                .toArray();
+                            }
+                        } else if (parameterType == String.class) {
+                            parameters[i] = (String) currentMetaInput;
```

Comment:
Will this work when we want to pass null as an argument?

Reply:
I have an initial check if the `currentMetaInput` is null [here](https://github.com/mongodb/mongo-jdbc-driver/pull/62/files/1c69105a80c2a68a611e7414604c01b16fb7d866#diff-e017435f177b34f6c79fba493b695e25515551ac3ac287cc8554c7257bababbbR163) (line 163 if the diff doesn't load).  Is that what you were referring to?  If null it will set the parameter[i] to `null` and continue.

Reply:
oh perfect, sounds good 👍